### DISCUSSION TITLE
Getting a list of QuantityTypes their units and there abbrivation

### DISCUSSION THREAD

**Discussion by DSRCFeroz** [angularsen/UnitsNet#877] [Q&A]
Hi, I am trying to create a list of the above-mentioned items as a list. 
something like 
{
Type:"Preassure",
Units:[
{UnitName: "pascal" , Abb: "Pa"}
{UnitName: "Atmosphere" , Abb: "ATM"}
]
}

can anyone give me a suggestion on how to get this list dynamically on runtime?

**Comment by angularsen** [comment]
Hi, take a look at `Quantity` and its static members like `Quantity.Infos` to get information about all quantities.
For information about a particular quantity, you can do `Pressure.Info`.

Did that help?

**Reply by DSRCFeroz** [reply]
@angularsen thanks for the quick reply.

I want to return a list of all the quantities and their unit and abbreviation

i have something like this :
` List<UnitsDto> _units = new List<UnitsDto>();
            QuantityType[] quantityTypes = Enum.GetValues(typeof(QuantityType)).Cast<QuantityType>().ToArray();
            foreach (QuantityType unitValue in quantityTypes)
            {
                if (unitValue != QuantityType.Undefined)
                {
                    QuantityInfo QtyInfo = Quantity.GetInfo(unitValue);
                    _units.Add(
                        new UnitsDto
                        {
                            Type = QtyInfo.Name,
                            Units = QtyInfo.UnitNames
                        });
                }
            }
            return _units;`

this gives me an output like:
  {
        "name": "Acceleration",
        "unitNames": [
            "CentimeterPerSecondSquared",
            "DecimeterPerSecondSquared",
            "FootPerSecondSquared",
            "InchPerSecondSquared",
            "KilometerPerSecondSquared",
            "KnotPerHour",
            "KnotPerMinute",
            "KnotPerSecond",
            "MeterPerSecondSquared",
            "MicrometerPerSecondSquared",
            "MillimeterPerSecondSquared",
            "MillistandardGravity",
            "NanometerPerSecondSquared",
            "StandardGravity"
        ]
    },
    {
        "name": "AmountOfSubstance",
        "unitNames": [
            "Centimole",
            "CentipoundMole",
            "Decimole",
            "DecipoundMole",
            "Kilomole",
            "KilopoundMole",
            "Megamole",
            "Micromole",
            "MicropoundMole",
            "Millimole",
            "MillipoundMole",
            "Mole",
            "Nanomole",
            "NanopoundMole",
            "PoundMole"
        ]
     ..., and so on.

is there any way to get abbreviations along with this?

**Comment by angularsen** [comment]
This seems to do something similar to what you're looking for:
```c#
Quantity
.Infos
.SelectMany(qi => qi.UnitInfos.Select(ui => new 
	{ 
		QuantityName = qi.Name,
		UnitName = ui.Name, 
		Abbreviations = string.Join("|", UnitAbbreviationsCache.Default.GetUnitAbbreviations(ui.Value.GetType(), Convert.ToInt32(ui.Value)))
	})
)
.GroupBy(x => x.QuantityName)
```

**Reply by DSRCFeroz** [reply]
That's brilliant....... with some slight changes based on my requirement it worked like a charm, Thank you so much.
Merry Christmas and happy new year.

**Reply by angularsen** [reply]
Happy to help, merry Christmas and happy new year to you too!
